pkg/controller/vsphere/util.go

```diff
@@ -252,3 +253,49 @@ func ProviderStatusFromRawExtension(rawExtension *runtime.RawExtension) (*machin
 func isNotFoundErr(err error) bool {
 	return err != nil && strings.HasSuffix(err.Error(), http.StatusText(http.StatusNotFound))
 }
+
+// podPredicate is a predicate function for filtering PodList
+type podPredicate func(corev1.Pod) bool
+
+// isTerminating is a predicate for determine pods in 'Terminating' state
+func isTerminating(p corev1.Pod) bool {
+	return p.DeletionTimestamp != nil
```

Comment:
I think terminating is a phase in a pod no? Isn't it in the status?

Reply:
No. It's not in the status.

Reply:
How does kubectl report that a pod is terminating? Doesn't kubectl get pod say `Terminating` when it's trying to remove a pod?

Reply:
>Doesn't kubectl get pod say Terminating when it's trying to remove a pod?

It looks into deletion timestamp, as I understand.

Reply:
https://github.com/kubernetes/kubernetes/blob/release-1.26/staging/src/k8s.io/kubectl/pkg/describe/describe.go#L789